### DISCUSSION TITLE
3.5: Improve side-by-side field vertical alignment

### DISCUSSION THREAD

**Discussion by evanwarner** [craftcms/cms#8867] [Ideas]
Opinion: side-by-side field alignment isn't optimal when the two fields have a different height.

Example 1: different length labels and only one wraps:

![craft-3_5-field-align](https://user-images.githubusercontent.com/6453842/88122907-9ab39680-cb7e-11ea-8eac-aa2b228aadc9.png)

Changing the parent wrapper to have `align-items: flex-end` makes this nicer:

![craft-3_5-field-align-fixed](https://user-images.githubusercontent.com/6453842/88122938-aef79380-cb7e-11ea-9369-d0d6be2c22ab.png)

---

Same thing when only one field has a description:

![craft-3_5-field-align-b](https://user-images.githubusercontent.com/6453842/88123787-7fe22180-cb80-11ea-9bb2-9fd0543bc16a.png)

vs:

![craft-3_5-field-align-b-fixed](https://user-images.githubusercontent.com/6453842/88123801-86709900-cb80-11ea-9bbd-1685aaf2f938.png)

**Comment by brandonkelly** [comment]
Considered that, but I’m not stoked about having the labels at different heights. Also played with moving the instruction text after the input, but that felt like it could lead to confusion, as the wording of the instruction text is currently all written assuming the current placement.

For now I’m going to leave it as-is, and maybe look into aligning the labels, instructions, and inputs together using subgrid once that’s a thing.

**Comment by evanwarner** [comment]
Yeah, agreed that instructions should stay above the input. I played around with keeping both labels and inputs aligned, but separating the label from the field created a bad cognitive gap:

![craft-3_5-field-align-flexcol](https://user-images.githubusercontent.com/6453842/88217981-3c38f780-cc14-11ea-8c01-fec0f4f8ab05.png)

Leaving it as-is until more thought and feedback has happened is the right choice.

**Comment by richhayler** [comment]
Could the field instructions be incorporated into a tooltip-style element, i.e. replace the text below the label with a “help/instructions” icon and when hovering over the icon display the instructions as a tooltip?

**Comment by brandonkelly** [comment]
@richhayler Possible but I think that would be a step in the wrong direction UX-wise.
